src/Components/App/App.tsx

```diff
@@ -24,14 +26,24 @@ const App: React.FC = () => {
   }
 
   return (
-    <div className="backGround">
-      <NavBar retrieveBooks={retrieveBooks} />
-      <main>
-        <h1 className="project-title">A Novel Idea</h1>
-        <BookCardContainer allBooks={books} oneBook={retrieveSingleBook}/>
-          <BookDetails />
-      </main>
-    </div>
+        <div className="backGround">
+          <main>
+            <h1 className="project-title">A Novel Idea</h1>
+            <Route exact path='/'>
+              <Home />
+            </Route>
+            <Route exact path='/books'>
+              <BookCardContainer allBooks={books} oneBook={retrieveSingleBook}/>
+              <NavBar retrieveBooks={retrieveBooks} />
+            </Route>
+            <Route exact path='/bookDetails'>
```

Comment:
Could you use an `id` here? Like.. `/books/:id`

This could be a bad idea, since I don't know your codebase fully. But something to consider!